src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1200,14 +1203,17 @@ private String copyStreamToBuildLog(InputStream stream, boolean info) {
                     error(line);
 
                     // Look for JVM version error in the line
-                    if (container && line.contains("JVMCFRE003")) {
-                        if (gradle) {
-                            // Gradle doesn't show errors in an obvious way, so use some formatting to make it stand out more
-                            error("***** [ ERROR ] ***** Java classes were compiled with a higher version of Java than the JVM in the container. To resolve this issue, set the source and target Java versions in your Gradle build to correspond to the Java version used in your Dockerfile or its parent image, then restart dev mode.");
-                        } else {
+                    alertOnServerError(line, "JVMCFRE003",
+                            "Java classes were compiled with a higher version of Java than the JVM in the container. To resolve this issue, set the source and target Java versions in your Gradle build to correspond to the Java version used in your Dockerfile or its parent image, then restart dev mode.",
                             // Maven project should be cleaned before restarting dev mode, otherwise compile does not realize Java version settings have changed
-                            error("Java classes were compiled with a higher version of Java than the JVM in the container. To resolve this issue, set the source and target Java versions in your Maven build to correspond to the Java version used in your Dockerfile or its parent image, then clean the project output and restart dev mode.");
-                        }
+                            "Java classes were compiled with a higher version of Java than the JVM in the container. To resolve this issue, set the source and target Java versions in your Maven build to correspond to the Java version used in your Dockerfile or its parent image, then clean the project output and restart dev mode.",
+                            false);
+
+                    // Look for features not available
+                    String errMsg = "Feature definitions were not found in the container. To install features to the container, specify 'RUN features.sh' in your Dockerfile. For an example of how to configure a Dockerfile, see https://github.com/OpenLiberty/ci.docker";
+                    if (!shownFeaturesShWarning) {
+                        boolean alerted = alertOnServerError(line, "CWWKF0001E", errMsg, errMsg, true);
+                        shownFeaturesShWarning = alerted;
```

Comment:
Just curious why setting `shownFeaturesShWarning` was done in two steps instead of one. Is this better practice?

Reply:
Good catch. Fixed.